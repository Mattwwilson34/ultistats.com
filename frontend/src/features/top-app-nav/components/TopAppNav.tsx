import * as React from 'react'
import AppBar from '@mui/material/AppBar'
import Box from '@mui/material/Box'
import Toolbar from '@mui/material/Toolbar'
import Typography from '@mui/material/Typography'
import Button from '@mui/material/Button'
import IconButton from '@mui/material/IconButton'
import logoutUser from '../../../utils/logout-user'

export default function TopAppNav(): React.ReactElement {
  return (
    <Box sx={{ flexGrow: 1, width: 1, marginBottom: 2 }}>
      <AppBar position="static">
        <Toolbar disableGutters>
          <IconButton color="inherit" aria-label="menu"></IconButton>
          <Typography variant="h6" component="div" sx={{ flexGrow: 1 }}>
            UltiStats
          </Typography>
          <Button color="inherit" onClick={logoutUser}>
            Logout
          </Button>
        </Toolbar>
      </AppBar>
    </Box>
  )
}
