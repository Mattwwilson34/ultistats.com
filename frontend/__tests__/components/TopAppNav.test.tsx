import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import TopAppNav from '../../src/features/top-app-nav'
import logoutUser from '../../src/utils/logout-user'

jest.mock('../../src/utils/logout-user')

describe('TopAppNav', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  describe('TopAppNav component', () => {
    it('matches snapshot', () => {
      const { asFragment } = render(<TopAppNav />)
      expect(asFragment()).toMatchSnapshot()
    })

    it('renders the app name', () => {
      const { getByText } = render(<TopAppNav />)
      const appName = getByText(/UltiStats/)
      expect(appName).toBeInTheDocument()
    })

    it('renders a logout button', () => {
      const { getByRole } = render(<TopAppNav />)
      const logoutButton = getByRole('button', { name: /Logout/ })
      expect(logoutButton).toBeInTheDocument()
    })

    it('calls logoutUser when the Logout button is clicked', () => {
      const { getByText } = render(<TopAppNav />)
      const logoutButton = getByText('Logout')
      fireEvent.click(logoutButton)
      expect(logoutUser).toHaveBeenCalled()
    })
  })
})
