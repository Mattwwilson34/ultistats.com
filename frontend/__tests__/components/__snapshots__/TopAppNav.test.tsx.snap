// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopAppNav TopAppNav component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-i9gxme"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
      >
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
        >
          UltiStats
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1y942vo-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Logout
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
